### ISSUE TITLE
Processus::run does not work on Windows

### ISSUE THREAD

**Issue by Hywan** [hoaproject/Console#79]
See https://github.com/hoaproject/Test/issues/83#issuecomment-281019576.

For the records, my message:

> Can you test those two examples and tell me which one works or not.
> 
> Example 1:
> 
> ```php
> $processus = new Hoa\Console\Processus('ls');
> $processus->open();
> echo $processus->readAll();
> ```
> 
> Example 2:
> 
> ```php
> $processus = new Hoa\Console\Processus('ls');
> $processus->on('output', function (Hoa\Event\Bucket $bucket) {
>     $data = $bucket->getData();
>     echo '> ', $data['line'], "\n";
> });
> $processus->run();
> ```
> 
> Thanks!

The reply:

> Test Nr.1:
> 
> ```
> CHANGELOG.md
> Client.php
> Connection
> Exception
> Node.php
> Processus.php
> README.md
> Server.php
> Socket.php
> Test
> Transport.php
> composer.json
> composer.lock
> test.sh
> vendor
> ```
> 
> Test Nr.2:
> 
> ```
> PHP Warning:  Invalid argument supplied for foreach() in D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\hoa\console\Processus.php on line 460
> 
> Warning: Invalid argument supplied for foreach() in D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\hoa\console\Processus.php on line 460
> ```

**Comment by Hywan** [comment]
Please @LukBukkit, can you remove the `@` in https://github.com/hoaproject/Console/blob/cc0a1bf3e90391d72434674290e05781b665cc4b/Processus.php#L401-L407 and `var_dump` the value of `$out`?

**Comment by LukBukkit** [comment]
Sure, i wasn't wheather you wanted to have the tests or the `test:run` command. So here are both:

**Test Nr.1:**
```
resource(34) of type (process) # var_dump($out)
array(3) {                                # var_dump($this->_pipes)
  [0]=>
  resource(31) of type (stream)
  [1]=>
  resource(32) of type (stream)
  [2]=>
  resource(33) of type (stream)
}
CHANGELOG.md
Client.php
Connection
Exception
Node.php
Processus.php
README.md
Server.php
Socket.php
Test
Transport.php
composer.json
composer.lock
test.sh
vendor
```

**Test Nr.2:**
```
NULL                                   # var_dump($this->_pipes) @ line 463
PHP Warning:  Invalid argument supplied for foreach() in D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\hoa\console\Processus.php on line 465

Warning: Invalid argument supplied for foreach() in D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\hoa\console\Processus.php on line 465
```

**`vendor/bin/hoa test:run`**
```
$ vendor/bin/hoa test:run
string(329) "D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\bin\atoum --autoloader-file D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\hoa\test\.autoloader.atoum.php --configurations D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\hoa\test\.atoum.php --force-terminal --max-children-number 2 --directories Test"
Hoa\Console\Processus::_open(): (8) Array to string conversion
in D:\lukas\Coding\PhpstormProjects\Hoaproject\Socket\vendor\hoa\console\Processus.php at line 406.
```

**Comment by Hywan** [comment]
Please, what version of Windows are you running?

**Comment by LukBukkit** [comment]
I'm running Windows 10 1607

Ivan Enderlin <notifications@github.com> schrieb am Mi., 22. Feb. 2017,
08:21:

**Comment by Hywan** [comment]
@LukBukkit Do you have an idea how I can test it myself?

**Comment by LukBukkit** [comment]
Does it work for you on Windows 10?

**Comment by Hywan** [comment]
I don't have Windows at all :-(.

**Comment by LukBukkit** [comment]
You can grab a Windows [Insider ISO](https://www.microsoft.com/en-us/software-download/windowsinsiderpreviewiso) or the version [1607](https://www.microsoft.com/en-us/software-download/windows10) from Microsoft. Then put it into a VM and install [XAMPP with PHP 7.0.15](https://www.apachefriends.org/download.html) or just [PHP 7.0.16 for Windows](http://windows.php.net/download#php-7.0) and last but not least [Git for Windows](https://git-scm.com/download/win) and maybe an editor of your choice. :)

**Comment by Hywan** [comment]
@LukBukkit Thanks!

Also, is `ls` working on Windows? Isn't it `dir`?

**Comment by LukBukkit** [comment]
Yes it's `dir` if you use the Windows cmd, but I use the Git Bash for my tests.

**Comment by Hywan** [comment]
@LukBukkit Really really really sorry for the too late reply. I have finally time to work on this issue. After 6h, I have a running Windows. Let's dig into this bug.

**Comment by Hywan** [comment]
@LukBukkit I guess we have an issue. I am not able to reproduce :-/. I have PHP 5.6 installed on Windows 10 (version 1607).

**Comment by Hywan** [comment]
ping ❤️?

**Comment by LukBukkit** [comment]
Sorry I was a little busy, I'll try to reproduce this and create a step for step documentation that show exactly how I broke it. I hope I can find time for this tomorrow.

**Comment by Hywan** [comment]
ping <3 ?

**Comment by LukBukkit** [comment]
Hello again,
sorry I'm currently a little bit busy. When I find time I'm going to try everything again and send you the results. 😉
